http/http_sink.go

```diff
@@ -129,7 +130,7 @@ func (h *HTTPSink) BatchConsume(msgs []interface{}, version int) {
 	}
 
 	//retry Execute till you succede based on retry config
-	status := h.retryExecute(h.conf.Method, url, headers, payload, responseCodeEvaluation)
+	status, _ := h.retryExecute(h.conf.Method, url, headers, payload, responseCodeEvaluation, math.MaxInt32)
```

Comment:
Why are we not processing error for batch here?

Reply:
For batch the implementation is not clear, 
We need to think what will we do if a single message fails in a batch as we going to sideline all the messages present in the batch or should we select the only message that has failed in the batch based on this logic for committing of the offsets also will change. 
So left it for now, may be this can taken as an enhancement at later stages.
Since we are using same retryExecute function for batch and non-batch defaulted to maxInt.

Reply:
I think we should still throw the error instead of ignoring it. It can cause unintended behaviour.